modules/tensor_mechanics/src/bcs/Pressure.C

```diff
@@ -31,17 +35,144 @@ Pressure::validParams()
 Pressure::Pressure(const InputParameters & parameters)
   : IntegratedBC(parameters),
     _component(getParam<unsigned int>("component")),
+    _ndisp(coupledComponents("displacements")),
     _factor(getParam<Real>("factor")),
     _function(isParamValid("function") ? &getFunction("function") : NULL),
     _postprocessor(isParamValid("postprocessor") ? &getPostprocessorValue("postprocessor") : NULL),
-    _alpha(getParam<Real>("alpha"))
+    _alpha(getParam<Real>("alpha")),
+    _fe_side(_assembly.getFEFace(_var.feType(), _sys.mesh().dimension())),
+    _q_dxi(nullptr),
+    _q_deta(nullptr),
+    _phi_dxi(nullptr),
+    _phi_deta(nullptr),
+    _qrule(_sys.mesh().dimension() - 1, _var.feType().default_quadrature_order()),
+    _use_displaced_mesh(getParam<bool>("use_displaced_mesh"))
 {
   if (_component > 2)
     mooseError("Invalid component given for ", name(), ": ", _component, ".\n");
+
+  for (unsigned int i = 0; i < _ndisp; ++i)
+    _disp_var.push_back(coupled("displacements", i));
+
+  _fe.resize(libMesh::n_threads());
+  const unsigned int dim = _sys.mesh().dimension() - 1;
+  for (unsigned int i = 0; i < libMesh::n_threads(); i++)
+  {
+    _fe[i] = FEBase::build(dim, _var.feType()).release();
```

Comment:
Why release and not use a unique ptr to hold the Fe? That's the preferred method over explicitly deleting.